src/utils/assert_op.jl

```diff
@@ -0,0 +1,46 @@
+"""
+Throw an `AssertionError` if `op(exp1, exp2)` is `false`.
+
+# Examples
+```
+julia> a = 3; b = 4;
+julia> @assert_op a == b
+ERROR: AssertionError: 3 is not isequal 4
+
+julia> @assert_op a + 3 == b + 4
+ERROR: AssertionError: 6 is not == 8
+
+julia> IS.@assert_op a isequal b
+ERROR: AssertionError: 3 is not isequal 4
+```
+"""
+macro assert_op(exp1, op, exp2)
+    assert_op(exp1, op, exp2)
+end
+
+macro assert_op(expr...)
+    assert_op(expr)
+end
+
+function assert_op(expr::Expr)
+    @assert expr.head == :call
+    @assert length(expr.args) == 3
+
+    assert_op(expr.args[2], expr.args[1], expr.args[3])
+end
+
+function assert_op(expr::Tuple{Expr})
+    @assert length(expr) == 1
+    assert_op(expr[1])
+end
+
+function assert_op(exp1, op, exp2)
+    return :(
+        if !$op($(esc(exp1)), $(esc(exp2)))
+            val1 = $(esc(exp1))
+            val2 = $(esc(exp2))
+            op_str = $(esc(op))
+            throw(AssertionError("$val1 is not $op_str $val2"))
```

Comment:
How about this?

```julia
throw(AssertionError("$op_str($val1, $val2) != true"))
```

Or

```julia
throw(AssertionError("!$op_str($val1, $val2)"))
```

Or

```julia
throw(AssertionError("$val1 !$op_str $val2"))
```

Reply:
Having the variable name in there seems useful though.

```julia
throw(AssertionError("Assertion failed between $exp1 and $exp2 using $op. $val1 $op_str $val2 is not true"))
```

Reply:
![Screen Shot 2020-11-12 at 10 23 53 AM](https://user-images.githubusercontent.com/1813121/98973590-2ad06f80-24d1-11eb-8dfa-e8f95f5f5862.png)

Reply:
Also, there's lots of corner cases this implementation doesn't handle. 

![Screen Shot 2020-11-12 at 10 26 05 AM](https://user-images.githubusercontent.com/1813121/98973826-7a16a000-24d1-11eb-8129-6f4f4b819931.png)

But I think that is fine, most people using this will be using just the 3 argument expression.

Reply:
Resolved in 7973a24.